crates/uv-workspace/src/dependency_groups.rs

```diff
@@ -154,6 +154,8 @@ pub enum DependencyGroupError {
     DevGroupInclude(GroupName),
     #[error("Detected a cycle in `dependency-groups`: {0}")]
     DependencyGroupCycle(Cycle),
+    #[error("Group `{0}` contains a Dependency Object Specifier, which is not supported by uv")]
+    DependencyObjectSpecifierNotSupported(GroupName),
```

Comment:
Can you make this take and print out the map? (Also I wonder if this should be less jargony...)

```suggestion
    #[error("Group `{0}` contains an unknown dependency object specifier: {1:?}")]
    DependencyObjectSpecifierNotSupported(GroupName, BTreeMap<String, String>),
```